x/tx/signing/aminojson/encoder_test.go

```diff
@@ -9,21 +9,64 @@ import (
 	"gotest.tools/v3/assert"
 )
 
-func TestCosmosBytesAsString(t *testing.T) {
+func TestCosmosInlineJSON(t *testing.T) {
 	cases := map[string]struct {
 		value      protoreflect.Value
 		wantErr    bool
 		wantOutput string
 	}{
-		"valid bytes - json": {
+		"supported type - valid JSON object": {
 			value:      protoreflect.ValueOfBytes([]byte(`{"test":"value"}`)),
 			wantErr:    false,
 			wantOutput: `{"test":"value"}`,
 		},
-		"valid bytes - string": {
-			value:      protoreflect.ValueOfBytes([]byte(`foo`)),
+		"supported type - valid JSON array": {
+			// spaces are normalized away
+			value:      protoreflect.ValueOfBytes([]byte(`[1,2,3]`)),
 			wantErr:    false,
-			wantOutput: `foo`,
+			wantOutput: `[1,2,3]`,
+		},
+		"supported type - valid JSON is not normalized": {
+			value:      protoreflect.ValueOfBytes([]byte(`[1, 2, 3]`)),
+			wantErr:    false,
+			wantOutput: `[1, 2, 3]`,
+		},
+		"supported type - valid JSON array (empty)": {
+			value:      protoreflect.ValueOfBytes([]byte(`[]`)),
+			wantErr:    false,
+			wantOutput: `[]`,
+		},
+		"supported type - valid JSON number": {
+			value:      protoreflect.ValueOfBytes([]byte(`43.72`)),
+			wantErr:    false,
+			wantOutput: `43.72`,
+		},
+		"supported type - valid JSON boolean": {
+			value:      protoreflect.ValueOfBytes([]byte(`true`)),
+			wantErr:    false,
+			wantOutput: `true`,
+		},
+		"supported type - valid JSON null": {
+			value:      protoreflect.ValueOfBytes([]byte(`null`)),
+			wantErr:    false,
+			wantOutput: `null`,
+		},
+		"supported type - valid JSON string": {
+			value:      protoreflect.ValueOfBytes([]byte(`"hey yo"`)),
+			wantErr:    false,
+			wantOutput: `"hey yo"`,
+		},
+		"supported type - invalid JSON": {
```

Comment:
since you're expected an error here, should the case include the name "supported type"?

Reply:
My understanding is this: The term "type" here refers to the data type. "ValueOfBytes" is a supported type, but bool and int64 are not. This highlights two levels of errors: one is a wrong type and the other one is wrong contents of the right type.

However, we can also rephrase those to failure reasons only. No strong opinion what is better.